plugins/outputs/zabbix/zabbix.go

```diff
@@ -0,0 +1,164 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package zabbix
+
+import (
+	_ "embed"
+	"sort"
+	"time"
+
+	"github.com/datadope-io/go-zabbix/v2"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// zabbixSender is an interface to send autoregister data to Zabbix.
+// It is implemented by Zabbix.Sender.
+// Created to be able to mock Zabbix.Sender in tests.
+type zabbixSender interface {
+	Send(packet *zabbix.Packet) (res zabbix.Response, err error)
+	SendMetrics(metrics []*zabbix.Metric) (resActive zabbix.Response, resTrapper zabbix.Response, err error)
+	RegisterHost(hostname string, hostMetadata string) error
+}
+
+// Zabbix allows pushing metrics to Zabbix software
+type Zabbix struct {
+	Address               string `toml:"address"`
+	AgentActive           bool   `toml:"agent_active"`
+	Prefix                string `toml:"prefix"`
+	SkipMeasurementPrefix bool   `toml:"skip_measurement_prefix"`
+
+	LLDSendInterval  config.Duration `toml:"lld_send_interval"`
+	LLDClearInterval config.Duration `toml:"lld_clear_interval"`
+	// lldHandler handles low level discovery data
+	lldHandler zabbixLLD
+	// lldLastSend store the last LLD send to known where to send it again
+	lldLastSend time.Time
+
+	Autoregister               string          `toml:"autoregister"`
+	AutoregisterResendInterval config.Duration `toml:"autoregister_resend_interval"`
+	// autoregisterLastSend stores the last time autoregister data was sent to Zabbix for each host.
+	autoregisterLastSend map[string]time.Time
+
+	Log telegraf.Logger `toml:"-"`
+
+	// sender is the interface to send data to Zabbix.
+	sender zabbixSender
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Zabbix) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect does nothing, Write() would initiate connection in each call.
+// Checking if Zabbix server is alive in this step does not allow Telegraf
+// to start if there is a temporal connection problem with the server.
+// Initilize LLD and autoregister maps. Copy config values to them. Configure Logger.
+func (z *Zabbix) Connect() error {
+	z.sender = zabbix.NewSender(z.Address)
+	// Initialize autoregisterLastSend map with size one, as the most common scenario is to have one host.
+	z.autoregisterLastSend = make(map[string]time.Time, 1)
+	z.lldLastSend = time.Now()
```

Comment:
This will make the plugin to wait for a full LLD-resend-period until the first bunch of LLD metrics is sent. Is this intended?

Reply:
No, it will send the first LLD metrics at `LLDSendInterval`.

Resend only affects for LLD not already sent.

Reply:
> No, it will send the first LLD metrics at LLDSendInterval.

That's what I meant, so it is intended. Thanks for the clarifications!